lib/ContractInteract/TokenHolder.js

```diff
@@ -0,0 +1,185 @@
+    }
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const args = [];
+    const contract = Contracts.getTokenHolder(auxiliaryWeb3);
+
+    return contract.deploy({
+      data: bin,
+      arguments: args
+    });
+  }
+
+  /**
+   * Authorizes a session. Expiration height should be greater than current block height.
+   *
+   * @param {String} sessionKey Session key address to authorize.
+   * @param {String} spendingLimit Spending limit of the session key.
+   * @param {String} expirationHeight Expiration height of the session key.
+   *
+   * @returns {String} Executable data for authorizing a session.
+   */
+  getAuthorizeSessionExecutableData(sessionKey, spendingLimit, expirationHeight) {
+    return this.contract.methods.authorizeSession(sessionKey, spendingLimit, expirationHeight).encodeABI();
+  }
+
+  /**
+   * Revokes session for the specified session key.
+   *
+   * @param {String} sessionKey Session key to revoke.
+   *
+   * @returns {String} Executable data to revoke a session.
+   */
+  getRevokeSessionExecutableData(sessionKey) {
+    return this.contract.methods.revokeSession(sessionKey).encodeABI();
+  }
+
+  /**
+   * Logout all authorized sessions.
+   *
+   * @returns {String} Executable data to logout sessions.
+   */
+  getLogoutExecutableData() {
+    return this.contract.methods.logout().encodeABI();
+  }
+
+  /**
+   * It is used to get call prefix of executeRule method in TokenHolder contract.
+   *
+   * @returns {String} Encoded signature of executeRule method.
+   */
+  getTokenHolderExecuteRuleCallPrefix() {
+    const executeRuleHash = this.auxiliaryWeb3.utils.soliditySha3(
+      'executeRule(address,bytes,uint256,uint8,bytes32,bytes32)'
+    );
+    const executeRuleCallPrefix = executeRuleHash.substring(0, 10);
+    return executeRuleCallPrefix;
+  }
+
+  /**
+   * It is used to execute executable data signed by a session key.
+   *
+   * @param {String} to The target contract address the transaction will be executed upon.
+   * @param {String} data The payload of a function to be executed in the target contract.
+   * @param {String} nonce The nonce of an session key that was used to sign the transaction.
+   * @param {String} r `r` part of the signature.
+   * @param {String} s `s` part of the signature.
+   * @param {String} v `v` part of the signature.
+   * @param {Object} txOptions Tx options.
+   *
+   * @returns {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async executeRule(to, data, nonce, r, s, v, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+    const txObject = await this.executeRuleRawTx(to, data, nonce, r, s, v);
```

Comment:
`await`  😨

Reply:
All rawTx method returns Promise now for consistency.